workshops/AndroidKotlinWorkshop/app/src/test/java/com/novoda/workshop/credentials/CredentialsPresenterTest.kt

```diff
@@ -0,0 +1,51 @@
+package com.novoda.workshop.credentials
+
+import com.nhaarman.mockitokotlin2.KArgumentCaptor
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.novoda.workshop.credentials.view.CredentialsView
+import org.junit.Before
+import org.junit.Test
+
+class CredentialsPresenterTest {
+    private val view: CredentialsView = mock()
+    private val navigator: CredentialsNavigator = mock()
+    private val presenter = CredentialsPresenter(navigator)
+    @Before
+    fun setUp() {
+        presenter.startPresenting(view)
+    }
+
+    @Test
+    fun `shows error when empty user name submitted`() {
+        val functionCapture = argumentCaptor<(String, String) -> Unit>()
+
+        capturedSubmitFunction(functionCapture).invoke("", "token")
```

Comment:
took me a while to understand what's up here.
Maybe `submitAction().invoke("", "token")`?